tracer_manager/tracer_manager.F90

```diff
@@ -1105,16 +1121,36 @@ subroutine set_tracer_profile(model, n, tracer, err_msg)
     select case (tracers(n1)%model)
       case (MODEL_ATMOS)
         multiplier = exp( log (top_value/surf_value) /numlevels)
-        tracer(:,:,1) = surf_value
-        do k = 2, size(tracer,3)
-          tracer(:,:,k) = tracer(:,:,k-1) * multiplier
-        enddo
+        select type (tracer)
+        type is (real(kind=r4_kind))
+          tracer(:,:,1) = surf_value
+          do k = 2, size(tracer,3)
+            tracer(:,:,k) = tracer(:,:,k-1) * multiplier
+          enddo
+        type is (real(kind=r8_kind))
+          tracer(:,:,1) = surf_value
+          do k = 2, size(tracer,3)
+            tracer(:,:,k) = tracer(:,:,k-1) * multiplier
+          enddo
+        class default
+          call mpp_error(FATAL, "set_tracer_profile : unsupported kind")
+        end select
       case (MODEL_OCEAN)
         multiplier = exp( log (bottom_value/surf_value) /numlevels)
-        tracer(:,:,size(tracer,3)) = surf_value
-        do k = size(tracer,3) - 1, 1, -1
-          tracer(:,:,k) = tracer(:,:,k+1) * multiplier
-        enddo
+        select type (tracer)
+        type is (real(kind=r4_kind))
+          tracer(:,:,size(tracer,3)) = surf_value
+          do k = size(tracer,3) - 1, 1, -1
+            tracer(:,:,k) = tracer(:,:,k+1) * multiplier
+          enddo
+        type is (real(kind=r8_kind))
+          tracer(:,:,size(tracer,3)) = surf_value
+          do k = size(tracer,3) - 1, 1, -1
+            tracer(:,:,k) = tracer(:,:,k+1) * multiplier
+          enddo
+        class default
+          call mpp_error(FATAL, "set_tracer_profile : unsupported kind")
```

Comment:
This error message should be more descriptive.  *"tracer is not one of the supported types of real(kind=4) or real(kind=8)."*